### PR TITLE
factor out actx init

### PR DESCRIPTION
**Pull request by matthiasdiener**
**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

**Review by MTCam — COMMENTED** [review]
I love this.  Maybe we don't need the `use_profiling` or the interface to query whether it is a profiling actx.   Maybe we should consider sticking actx utils into a separate module like array_context.py so that we import from mirgecom.array_context.

File: mirgecom/simutil.py [hunk 1]
```diff
@@ -1129,6 +1137,70 @@ def configurate(config_key, config_object=None, default_value=None):
...
+    from arraycontext import PyOpenCLArrayContext
+    return issubclass(actx_class, PyOpenCLArrayContext)
+
+
+def actx_class_is_profiling(actx_class) -> bool:
+    from mirgecom.profiling import PyOpenCLProfilingArrayContext
+    return issubclass(actx_class, PyOpenCLProfilingArrayContext)
+
+
+def initialize_actx(actx_class: Type[ArrayContext], comm: Optional["Comm"]) -> Tuple[ArrayContext, cl.Context, cl.CommandQueue, cl.tools.AllocatorBase]:
```
Comment (majosm) [inline mirgecom/simutil.py:1170]: Maybe only return `actx`, so it's compatible with non-CL array contexts (numpy?)? It looks like `context`, `queue`, and `allocator` are all attributes of `actx` for pyopencl/pytato, so presumably they could be retrieved in the driver that way if needed.

**Review by majosm — COMMENTED** [review]
Just one thought, otherwise I like it. 👍

File: mirgecom/simutil.py (hunk 1 above)
Comment (matthiasdiener) [reply mirgecom/simutil.py]: Thanks, I think that makes sense. I changed it in [6578353](https://github.com/illinois-ceesd/mirgecom/pull/918/commits/65783536c02c3a01bb3d184474d11c48b85e3e87)

**Review by MTCam — APPROVED** [review]
This is great.

**Pull request merged**